src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,124 @@
-import { useState } from 'react';
+import { FC, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void
+}
+
+export const NewMovie: FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const requiredTextFields = (title && imgUrl && imdbUrl && imdbId) === '';
+
+  const handleTitle = (value: string) => {
+    setTitle(value);
+  };
+
+  const handleDescription = (value: string) => {
+    setDescription(value);
+  };
+
+  const handleImgUrl = (value: string) => {
+    setImgUrl(value);
+  };
+
+  const handleImdbUrl = (value: string) => {
+    setImdbUrl(value);
+  };
+
+  const handleImdbId = (value: string) => {
+    setImdbId(value);
+  };
+
+  const clearForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setCount(prevCount => prevCount + 1);
+  };
+
+  const submitHandler = () => {
+    const movie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(movie);
+
+    clearForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+        submitHandler();
+      }}
```

Comment:
You have a handler for this event. Use it correctly 

```suggestion
      onSubmit={handlerSubmit}
```